### PR TITLE
COL-1146, preview-service can handle standard S3 URIs of format: s3://my-bucket/my-object-key

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/COL-1146

This will allow SuiteC to pass along s3 URIs (as stored in suitec db) which include bucket and objectKey info.

### CODE REVIEW

**Comment by johncrossman** [comment]
Sister PR in SuiteC land: https://github.com/ets-berkeley-edu/suitec/pull/607

File: lib/core/storage.js [hunk 1]
```diff
@@ -106,3 +107,67 @@ var store = module.exports.store = function(file, callback) {
...
+  return isS3Uri(s3Uri) ? s3Uri.split('/')[2] : null;
+};
+
+/**
+ * @param  {String}     s3Uri          URI of object in S3 (e.g., s3://my-bucket/my-object-key)
+ * @return {String}                    Strictly the S3 Object Key
+ */
+var extractS3ObjectKey = function(s3Uri) {
+  return isS3Uri(s3Uri) ? s3Uri.split('/').slice(3).join('/') : null;
+};
```
Comment (johncrossman) [inline lib/core/storage.js]: @paulkerschen The amendment made in linked PR will be made above, too. Stand by.

**Comment by johncrossman** [comment]
Amendment is in. Verified with 
download

**Comment by paulkerschen** [comment]
All right, preview-service PR, go play with your sister.

**Pull request merged**